### PR TITLE
Prevent rspec from being executed in any env other than test

### PR DESCRIPTION
**Pull request by rafatower**
Equivalent to https://github.com/CartoDB/cartodb-central/pull/3045

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 # This file is copied to spec/ when you run 'rails generate rspec:install'
 ENV['RAILS_ENV'] ||= 'test'
```
Comment (amiedes) [inline spec/spec_helper.rb:10]: Should we remove this line also? That way we avoid test from being executed if `RAILS_ENV` is empty

File: spec/spec_helper_min.rb [hunk 2]
```diff
@@ -8,6 +8,8 @@
 require './spec/support/shared_entities_spec_helper'
 
 ENV['RAILS_ENV'] ||= 'test'
```
Comment (amiedes) [inline spec/spec_helper_min.rb:10]: Should we remove this line also? That way we avoid test from being executed if `RAILS_ENV` is empty

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -8,6 +8,8 @@
...
+raise %(Cannot run tests in an env other than 'test', RAILS_ENV=#{ENV['RAILS_ENV']}) unless ENV['RAILS_ENV'] == 'test'
```
Comment (amiedes) [inline spec/spec_helper.rb:11]: What about using `Rails.env` / `Rails.env.test?` instead of directly using `ENV['RAILS_ENV']`?

File: spec/spec_helper_min.rb [hunk 4]
```diff
@@ -8,6 +8,8 @@
...
+raise %(Cannot run tests in an env other than 'test', RAILS_ENV=#{ENV['RAILS_ENV']}) unless ENV['RAILS_ENV'] == 'test'
```
Comment (amiedes) [inline spec/spec_helper_min.rb:11]: What about using `Rails.env` / `Rails.env.test?` instead of directly using `ENV['RAILS_ENV']`?

**Review by amiedes — APPROVED** [review]
Just a couple of minor (optional) comments

File: spec/spec_helper.rb (hunk 1 above)
Comment (rafatower) [reply spec/spec_helper.rb:10]: That may depend on the CI scripts, but lets give it a try!

Comment (ibrahimupm) [reply spec/spec_helper.rb]: test
